test/unit/unicode_spec.js

```diff
@@ -0,0 +1,130 @@
+/* globals describe, it, expect, beforeAll, afterAll, mapSpecialUnicodeValues,
+           getUnicodeForGlyph, getGlyphsUnicode, getDingbatsGlyphsUnicode,
+           getUnicodeRangeFor, getNormalizedUnicodes, reverseIfRtl */
+
+'use strict';
+
+describe('unicode', function () {
+  describe('mapSpecialUnicodeValues', function () {
+    it('should not re-map normal Unicode values', function () {
+      // A
+      expect(mapSpecialUnicodeValues(0x0041)).toEqual(0x0041);
+      // fi
+      expect(mapSpecialUnicodeValues(0xFB01)).toEqual(0xFB01);
+    });
+
+    it('should re-map special Unicode values', function () {
+      // copyrightsans => copyright
+      expect(mapSpecialUnicodeValues(0xF8E9)).toEqual(0x00A9);
+      // Private Use Area characters
+      expect(mapSpecialUnicodeValues(0xFFFF)).toEqual(0);
+    });
+  });
+
+  describe('getUnicodeForGlyph', function () {
+    var standardMap, dingbatsMap;
+
+    beforeAll(function (done) {
+      standardMap = getGlyphsUnicode();
+      dingbatsMap = getDingbatsGlyphsUnicode();
+      done();
+    });
+
+    afterAll(function () {
+      standardMap = dingbatsMap = null;
+    });
+
+    it('should get Unicode values for valid glyph names', function () {
+      expect(getUnicodeForGlyph('A', standardMap)).toEqual(0x0041);
+      expect(getUnicodeForGlyph('a1', dingbatsMap)).toEqual(0x2701);
+    });
+
+    it('should recover Unicode values from uniXXXX/uXXXX{XX} glyph names',
+        function () {
+      expect(getUnicodeForGlyph('uni0041', standardMap)).toEqual(0x0041);
+      expect(getUnicodeForGlyph('u0041', standardMap)).toEqual(0x0041);
+
+      expect(getUnicodeForGlyph('uni2701', dingbatsMap)).toEqual(0x2701);
+      expect(getUnicodeForGlyph('u2701', dingbatsMap)).toEqual(0x2701);
+    });
+
+    it('should not get Unicode values for invalid glyph names', function () {
+      expect(getUnicodeForGlyph('Qwerty', standardMap)).toEqual(-1);
+      expect(getUnicodeForGlyph('Qwerty', dingbatsMap)).toEqual(-1);
+    });
+  });
+
+  describe('getUnicodeRangeFor', function () {
+    it('should get correct Unicode range', function () {
+      // A (Basic Latin)
+      expect(getUnicodeRangeFor(0x0041)).toEqual(0);
+      // fi (Alphabetic Presentation Forms)
+      expect(getUnicodeRangeFor(0xFB01)).toEqual(62);
+    });
+
+    it('should not get a valid Unicode range', function () {
```

Comment:
> Did you mean "an invalid Unicode range"?

Not really, since it'd be an error if we actually got a Unicode range here. Would the following clear up possible confusion: `it('should not get a Unicode range', function () {`